docs/eBpfExtensions.md

```diff
@@ -1,42 +1,207 @@
 # eBPF extensions
 
-## Overview
+## 1 Overview
+An "eBPF extension" is a Windows kernel driver or component that implements eBPF hooks or helper functions. The design of eBPF for Windows is
+such that an extension providing implementation for hooks and helper functions can be developed and deployed
+without the need to modify either the eBPF execution context or the eBPF verifier.
 
-The eBPF for Windows project is designed to permit anyone to add new hooks, program types or helper functions,
-without the need to modify either the eBPF execution context or the eBPF verifier.  We use the term "eBPF extension"
-to mean a driver or component that exposes new eBPF hooks or helper functions.
+## 1.1 Windows Network Module Registrar
 The eBPF for Windows project uses the
-[Network Module Registrar (NMR)](https://docs.microsoft.com/en-us/windows-hardware/drivers/network/network-module-registrar2)
-to decouple the eBPF extensions from the core eBPF for Windows framework.  In the NMR architecture,
+[Network Module Registrar (NMR)](https://docs.microsoft.com/en-us/windows-hardware/drivers/network/introduction-to-the-network-module-registrar)
+feature in Windows to develop eBPF extensions that are decoupled from the core eBPF for Windows framework.  NMR facilitates software
+modules to be attached to each other. The
 [Network Programming Interface (NPI)](https://docs.microsoft.com/en-us/windows-hardware/drivers/network/network-programming-interface)
-contracts are identified by a GUID.
+defines the contract between the modules.  The eBPF extensions would implement the
+[Provider Modules](https://docs.microsoft.com/en-us/windows-hardware/drivers/network/provider-module) for two types of NPIs as discussed in detail below.
+Even though the NMR component and the relevant APIs have the word "Network" in them, these software modules
+are completely agnostic of Windows Networking stack and non-networking eBPF extensions can be developed using these.
 
-## NPI Contract for eBPF Program Information
+## 1.2 Prerequisites
+The audience is assumed to be familiar with [NMR Architecture](https://docs.microsoft.com/en-us/windows-hardware/drivers/network/architecture-overview) and
+the various aspects of developing NMR modules as documented [here](https://docs.microsoft.com/en-us/windows-hardware/drivers/network/network-module-registrar2).
```

Comment:
```suggestion
the various aspects of developing NMR modules as described in [NMR documentation](https://docs.microsoft.com/en-us/windows-hardware/drivers/network/network-module-registrar2).
```
Avoid "here" for accessibility reasons, as explained at https://www.w3.org/TR/WCAG10-HTML-TECHS/#link-text